### PR TITLE
Metadata adapter interface

### PR DESCRIPTION
**Pull request by jkusa**
Resolves #

## Description

Refactor Metadata Adapters

## Proposed Changes

- Generate deterministic metadata descriptions
- Create NaviMetadataAdapter Interface
- Rename `fetchMetadata` to `fetchById`
- Move logic to fetch all metadata from service to adapter
- Move adapters to `adapter:metadata` namespace

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/tests/unit/adapters/metadata/bard-test.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import BardMetadataAdapter from 'navi-data/adapters/metadata/bard';
+//@ts-ignore
+import { setupMirage } from 'ember-cli-mirage/test-support';
```
Comment (kevinhinterlong) [inline packages/data/tests/unit/adapters/metadata/bard-test.ts:5]:
```suggestion
import { setupMirage } from 'ember-cli-mirage/test-support';
import { TestContext } from 'ember-test-helpers';
```

File: packages/data/tests/unit/adapters/metadata/bard-test.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+let Adapter: BardMetadataAdapter;
+
+module('Unit | Adapter | metadata/bard', function(hooks) {
+  setupTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(function(this: TODO) {
```
Comment (kevinhinterlong) [inline packages/data/tests/unit/adapters/metadata/bard-test.ts:13]:
```suggestion
  hooks.beforeEach(function(this: TestContext) {
```

File: packages/data/tests/unit/adapters/metadata/elide-test.ts [hunk 3]
```diff
@@ -1,32 +1,39 @@
...
 import { print } from 'graphql/language/printer';
 import scenario from 'dummy/mirage/scenarios/graphql';
+import ElideMetadataAdapter from 'navi-data/adapters/metadata/elide';
 
 // Apollo includes this field in its queries, but we don't want to compare against that in our tests
-const removeTypeNameField = query => query.replace(/\n\s*__typename/g, '').trim();
+const removeTypeNameField = (query: string) => query.replace(/\n\s*__typename/g, '').trim();
 
-module('Unit | Elide Metadata Adapter', function(hooks) {
+type TestContext = {
```
Comment (kevinhinterlong) [inline packages/data/tests/unit/adapters/metadata/elide-test.ts:13]: Could extend actual TestContext

Comment (jkusa) [reply packages/data/tests/unit/adapters/metadata/elide-test.ts:13]: Nice, that wasn't on my radar 👍

File: packages/data/tests/unit/adapters/metadata/bard-test.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    Adapter = this.owner.lookup('adapter:metadata/bard');
+  });
+
+  test('_buildURLPath', function(assert) {
+    assert.equal(
+      Adapter['buildURLPath']('table', '', {}),
```
Comment (kevinhinterlong) [inline packages/data/tests/unit/adapters/metadata/bard-test.ts:19]:
```suggestion
      Adapter.buildURLPath('table', '', {}),
```

Comment (jkusa) [reply packages/data/tests/unit/adapters/metadata/bard-test.ts:19]: @kevinhinterlong `buildURLPath` is private. Not sure if there is a better way to do this.

**Pull request merged**